### PR TITLE
live-server: init at 0.7.0

### PR DESCRIPTION
**Pull request by philiptaron**
## Description of changes

https://github.com/lomirus/live-server. It's a simple live server.

This is the "Rust alternative to live-server" referred to in [`pkgs/tools/nix/web-devmode.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/nix/web-devmode.nix#L74).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/li/live-server/package.nix
Comment (afh) [inline pkgs/by-name/li/live-server/package.nix:3]:
```suggestion
  lib,
  stdenv,
  rustPlatform,
  darwin,
```

File: pkgs/by-name/li/live-server/package.nix
Comment (afh) [inline pkgs/by-name/li/live-server/package.nix:24]:
```suggestion
  buildInputs = [ openssl ]
    ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
    CoreServices SystemConfiguration
  ]);
```

File: pkgs/by-name/li/live-server/package.nix
Comment (afh) [inline pkgs/by-name/li/live-server/package.nix:33]:
```suggestion
    platforms = platforms.linux ++ platforms.darwin;
```

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for this, @philiptaron, find below a few suggestions that enable building on darwin. Please let me know if there are any questions regarding these…

**Comment by philiptaron** [comment]
@afh, want to maintain this on macOS? 😁 Thanks for the suggestions -- I'll apply them.

**Comment by afh** [comment]
Thanks for the offer, @philiptaron, very much appreciated, yet I don't feel familiar enough with `buildRustPackage` and Rust in general to be comfortable with taking on a maintainer role for this package. Feel free to ping me directly in the future or reach out to the [darwin-maintainers](http://github.com/orgs/NixOS/teams/darwin-maintainers) team though 🙂

**Comment by afh** [comment]
Result of `nixpkgs-review pr 330279` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>live-server</li>
  </ul>
</details>

**Review by afh — APPROVED** [review]
Thanks for this, @philiptaron, from my end this looks good and thanks for accepting the suggestions adding support for darwin, much appreciated! 👍

**Review by pbsds — APPROVED** [review]
waiting for CI

**Comment by afh** [comment]
Thanks for merging this, @pbsds, very helpful 👍

**Pull request merged**
